### ISSUE TITLE
error.invalidAttributes does not contain the name of the field as specified in the model

### ISSUE THREAD

**Issue by micha-LEAP** [balderdashy/sails-mysql#187]
For example, I have a user model with attributes: 
```javascript
{
  username  : { type: 'string', unique: true },
  email     : { type: 'email',  unique: true },
  passports : { collection: 'Passport', via: 'user' },
}
```
If attempting to create a record with an existing email address the error object has the following `invalidAttributes`:
```javascript
{ 
  IX_UserIdentity_email: [ 
    { 
      value: 'micha.xxxxx@xxxxxxx.xxxxx',
      rule: 'unique',
      message: 'A record with that `IX_UserIdentity_email` already exists (`micha.xxxxx@xxxxxxx.xxxxx`).' 
    } 
  ]
}
```
I believe the attribute `IX_UserIdentity_email` should actually be `email` since that is the model attribute which failed validation. 

In the passport library, since this attribute name (`IX_UserIdentity_email`) does not match the model's attribute, the wrong error message is presented.

**Comment by devinivy** [comment]
I think this is a reasonable request.  Would love some input @tjwebb @dmarcelino

**Comment by dmarcelino** [comment]
Yeap, sounds reasonable. :+1: 

On a related note, we probably should standardise this kind of error messages across adapters, I've just checked sails-orientdb and it throws whatever OrientDB throws. This would probably be something for Waterline 1.0.0 :)

**Comment by sailsbot** [comment]
Thanks for posting, @micha-LEAP.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
